server/controllers/request.js

```diff
@@ -0,0 +1,69 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /requests
+// @desc list of requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const user = req.body.userId;
+
+  let requests;
+  if (user) {
+    requests = await Request.find({
+      $or: [{ userId: user }, { sitterId: user }],
+    });
+  }
+
+  if (!user) {
+    res.status(404);
+    throw new Error("No quest found.");
+  }
```

Comment:
II understand the idea here, but we do have middleware available to check if the user is authenticated.

Reply:
- I added middleware "protect" to request routes
- I also modified GET's responses to support front-end data rendering.